coreos-base/coreos-metadata/coreos-metadata-9999.ebuild

```diff
@@ -1,36 +1,204 @@
-# Copyright (c) 2015 CoreOS, Inc.. All rights reserved.
+# Copyright (c) 2017 CoreOS, Inc.. All rights reserved.
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
-CROS_WORKON_PROJECT="coreos/coreos-metadata"
-CROS_WORKON_LOCALNAME="coreos-metadata"
-CROS_WORKON_REPO="git://github.com"
-COREOS_GO_PACKAGE="github.com/coreos/coreos-metadata"
-inherit coreos-go cros-workon systemd
-
-if [[ "${PV}" == 9999 ]]; then
-	KEYWORDS="~amd64 ~arm64"
-else
-	CROS_WORKON_COMMIT="8f2578dc5231e27f05e8d105d39c03c057946862" # v0.14.0
-	KEYWORDS="amd64 arm64"
-fi
-
-DESCRIPTION="A simple cloud-provider metadata agent"
-HOMEPAGE="https://github.com/coreos/coreos-metadata"
-SRC_URI=""
+EAPI=6
+
+inherit systemd eutils coreos-cargo
 
+DESCRIPTION="a simple tool for collecting instance metadata from various providers"
+HOMEPAGE="https://github.com/coreos/coreos-metadata"
+KEYWORDS="amd64 arm64"
 LICENSE="Apache-2.0"
 SLOT="0"
-IUSE=""
+VERSION="47390b8f433c7e4a6be305c3460f7f75cfb46105"
```

Comment:
Should we just keep the `cros_workon` setup for this package?  (Or add it for `update-ssh-keys`?)

@euank Do you have a preference what repos are added to the manifest?

Reply:
I tried to add `cros_workon`, but it breaks because both `cros_workon` and the `cargo` eclass try and create a `src_unpack`, and the last one seems to win

Reply:
I pushed a patch onto this branch that uses `cros_workon` for `update-ssh-keys`.  (Feel free to force-push it off again; I just figured this was the easiest way to demonstrate a patch.)